microscopic-monks/primal/engine/sprite.py

```diff
@@ -0,0 +1,228 @@
+        self.bg_rect.texture.mag_filter = 'nearest'
+        self.bg_rect.texture.min_filter = 'nearest'
+
+    def get_source(self) -> str:
+        return self.bg_rect.source
+
+    def set_position(self, pos: Tuple[float, float]):
+        self.pos = pos
+        self.bg_rect.pos = pos
+
+    def set_size(self, size: Tuple[float, float]):
+        self.size = size
+        self.bg_rect.size = size
+
+    def get_size(self) -> Tuple[float, float]:
+        return self.bg_rect.size
+
+    def get_position(self) -> Tuple[float, float]:
+        return self.pos
+
+    def get_center(self):
+        return self.pos[0] + self.size[0] / 2, self.pos[1] + self.size[1] / 2
+
+    def draw(self, canvas: Union[RenderContext, InstructionGroup]):
+        canvas.add(self.bg_rect)
+
+
+class ColorSprite(Sprite):
+    def __init__(self, image: Union[str, None], pos: Tuple[float, float],
+                 size: Tuple[float, float], color: Tuple[float, float, float, float], **kwargs):
+        super().__init__(image, pos, size, **kwargs)
+
+        self.color = Color(*color)
+
+    def draw(self, canvas: Union[RenderContext, InstructionGroup]):
+        canvas.add(self.color)
+        canvas.add(self.bg_rect)
+        canvas.add(Color(1, 1, 1, 1))
+
+    def set_alpha(self, alpha: float):
+        self.color.a = alpha
+
+
+class RotatableSprite(Sprite):
+    def __init__(self, image: str, pos: Tuple[float, float],
+                 size: Tuple[float, float], angle: float, **kwargs) -> None:
+        super().__init__(image, pos, size, **kwargs)
+
+        self.rotate = Rotate(angle=angle, origin=self.get_center())
+
+    def set_angle(self, angle: float):
+        self.rotate.angle = angle
+
+    def get_rotation(self) -> float:
+        return self.rotate.angle
+
+    def draw(self, canvas: Union[RenderContext, InstructionGroup]):
+        canvas.add(PushMatrix())
+        canvas.add(self.rotate)
+        canvas.add(self.bg_rect)
+        canvas.add(PopMatrix())
+
+
+class Player(RotatableSprite):
+    SPEED = 300
+
+    def __init__(self, image: str, pos: Tuple[float, float],
+                 size: Tuple[float, float], angle: float, **kwargs):
+        super().__init__(image, pos, size, angle, **kwargs)
+        color = (0, 0, 0, 1)
+        self.timer = 0
+
+        self.in_hand = ColorSprite(None, (0, 0), (20, 20), (1, 1, 1, 1))
+        self.in_hand.set_alpha(0)
+        self.left_size = (20, size[1] * 0.3)
+        self.right_size = (20, size[1] * 0.3)
+        self.left = ColorSprite(None, (0, 0), self.left_size, color)
+        self.right = ColorSprite(None, (0, 0), self.right_size, color)
+
+    def set_rotation(self, n: Tuple[float, float]):
+        pos = self.pos
+        size = self.size
+        self.rotate.origin = pos[0] + size[0] / 2, pos[1] + size[1] / 2
+
+        x, y = n
+        d = math.atan2(y - Window.size[1] / 2, x - Window.size[0] / 2) * 180 / math.pi
+        self.rotate.angle = d
```

Comment:
I'd advocate against the usage of one letter variables here. In particularly, it's not clear what `n` and `d` represent here. A comment would also be helpful to describe the purpose of `d = math.atan2(y - Window.size[1] / 2, x - Window.size[0] / 2) * 180 / math.pi`.